src/Microsoft.Extensions.Configuration.AzureAppConfiguration/Extensions/TimeSpanExtensions.cs

```diff
@@ -72,5 +80,107 @@ public static TimeSpan CalculateBackoffDuration(this TimeSpan minDuration, TimeS
 
             return TimeSpan.FromMilliseconds(minDuration.TotalMilliseconds + new Random().NextDouble() * (calculatedMilliseconds - minDuration.TotalMilliseconds));
         }
+
+        /// <summary>
+        /// This method calculates the jittered exponential backoff duration for the configuration store after a failure
+        /// during startup which lies between <paramref name="minDuration"/> and <paramref name="maxDuration"/>.
+        /// </summary>
+        /// <param name="minDuration">The minimum duration to retry after.</param>
+        /// <param name="maxDuration">The maximum duration to retry after.</param>
+        /// <param name="attempts">The number of attempts made to the configuration store after the fixed retry period.</param>
+        /// <param name="startupTimeElapsed">The time elapsed since the current startup began.</param>
+        /// <returns>The backoff duration before retrying a request to the configuration store or replica again.</returns>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// An exception is thrown when <paramref name="attempts"/> is less than 1.
+        /// </exception>
+        public static TimeSpan CalculateExponentialStartupBackoffDuration(this TimeSpan minDuration, TimeSpan maxDuration, int attempts, TimeSpan startupTimeElapsed)
+        {
+            if (attempts < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(attempts), attempts, "The number of attempts should not be less than 1.");
+            }
+
+            if (attempts == 1)
+            {
+                return minDuration;
+            }
+
+            //
+            // IMPORTANT: This can overflow
+            double calculatedMilliseconds = Math.Max(1, minDuration.TotalMilliseconds) * ((long)1 << Math.Min(attempts, MaxAttempts));
+
+            if (calculatedMilliseconds > maxDuration.TotalMilliseconds ||
+                    calculatedMilliseconds <= 0 /*overflow*/)
+            {
+                calculatedMilliseconds = maxDuration.TotalMilliseconds;
+            }
+
+            TimeSpan maxBackoffDuration = CalculateMaxStartupBackoffDuration(startupTimeElapsed);
```

Comment:
The maxBackoffDuration variable shouldn't be needed. The max backoff in this case should always be our 10 min max backoff.

Reply:
What's the point of the maxDuration parameter then?

Reply:
I agree, I missed some things here. Zhenlan suggested just using `CalculateBackoffDuration` and adjusting that method to use jitter instead of the randomizing it does in the last line. At this point we're using the same backoff duration values, so unless there's a reason refresh is different from startup retries then I think it makes sense.

Reply:
A single method will be better. That's your suggestion right?

Reply:
Right, just one method and removing the `CalculateExponentialStartupBackoffDuration` method

Reply:
Perfect, let's do it.

Reply:
Just update that implementation with this one should be good right?